packages/react-ui/src/ui/utils/withDefaults.tsx

```
@@ -0,0 +1,35 @@
+  type TComponentProps = typeof defaultProps;
+  const getProps = withDefaultProps<TComponentProps>(defaultProps);
+  const WithDefaults = forwardRef<ComponentRef<TComponent>, TComponentProps>(
+    (props, ref) => {
+      const ComponentAsAny = Component as any;
+      return <ComponentAsAny {...getProps(props as any)} ref={ref} />;
+    },
+  );
+  WithDefaults.displayName =
+    "withDefaults(" +
+    (typeof Component === "string" ? Component : Component.displayName) +
+    ")";
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)